include/aspect/postprocess/core_statistics.h

```diff
@@ -0,0 +1,107 @@
+/*
+  Copyright (C) 2011, 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_postprocess_core_statistics_h
+#define _aspect_postprocess_core_statistics_h
+
+#include <aspect/postprocess/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/boundary_temperature/dynamic_core.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+
+    /**
+     * A postprocessor that computes some statistics about the dynamic core.
+     *
+     * @ingroup Postprocessing
+     */
+    template <int dim>
+    class CoreStatistics : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        CoreStatistics ();
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+        virtual
+        std::pair<std::string,std::string>
+        execute (TableHandler &statistics);
+
+        /**
+         * Export core data stored in this object. Doing this only because the boundary
+         * temperature doesn't allowed to store restart data there. So we store the data needed
+         * for restart here and exported to boundary temperature object if required.
+         */
+        struct  BoundaryTemperature::internal::CoreData
+      get_core_data() const;
```

Comment:
the same applies as above -- I think you probably want to return a `const` reference